torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1143,18 +1143,89 @@ void SpecialPostProcess(Node* n) {
       // onnx Sequence type requires element type to be set.
       // If the list to insert is empty, we set the elem type by
       // looking at the tensor being inserted.
-      auto list_node = n->input(0)->node();
+      auto seq_node = n->input(0)->node();
       auto t_node = n->input(1)->node();
-      if (!list_node || list_node->kind() != prim::ListConstruct ||
-          list_node->inputs().size() != 0) {
-        break;
-      }
+      auto t_type = n->input(1)->type()->cast<TensorType>();
+
+      auto update_sequence_empty_dtype = [](Node* n, TensorTypePtr t_type) {
+        TORCH_INTERNAL_ASSERT(n && n->kind() == ::c10::onnx::SequenceEmpty);
+        TORCH_INTERNAL_ASSERT(t_type->scalarType().has_value());
```

Comment:
We can't take the has_value() method if t_type is null. So should we also need to check nullptr for t_type？

Reply:
nice catch